unified-codes/apps/data-loader/src/app/v2/DataParser.ts

```diff
@@ -200,403 +185,83 @@ export class DataParser {
-            }
-          }
-        }
+        const productCode = productDefinition[0].code;
+        // If category code exists, and product code exists
+        if (uc1 && productCode) {
 
-        // If dose strength code exists...
-        if (uc6) {
-          // and unit code exists...
-          if (uc7) {
-            // link dose strength to unit.
-            if (!this.graph[uc6].children.map((child) => child.code).includes(uc7)) {
-              this.graph[uc6].children.push({ code: uc7 });
-              console.log(`INFO: Linked dose strength with code ${uc6} to unit with code ${uc7}`);
-            }
+          // link category to product.
+          if (!this.graph[uc1].children.map((child) => child.code).includes(productCode)) {
+            this.graph[uc1].children.push({ code: productCode });
+            console.log(`INFO: Linked category with code ${uc1} to product with code ${productCode}`);
           }
         }
 
-        // If dose qualifier code exists...
-        if (uc5) {
-          // and strength code exists...
-          if (uc6) {
-            // link dose qualification to strength.
-            if (!this.graph[uc5].children.map((child) => child.code).includes(uc6)) {
-              this.graph[uc5].children.push({ code: uc6 });
+        // Iterate through and assign children to parents 
+        // This starts at the top of the graph and for each node, finds the closest lower level node that has a code and name
+        // This avoids having to manually define each combination of 'levels' that are able to be connected 
+        let parentIndex = 0;
+        let childIndex = 1;
+        console.log(productDefinition)
+        while (childIndex < productDefinition.length) {
+          const parent = productDefinition[parentIndex];
+          const child = productDefinition[childIndex];
+          if (child.name && child.code) {
+            if (!this.graph[parent.code].children.map((child) => child.code).includes(child.code)) {
+              this.graph[parent.code].children.push(child);
               console.log(
-                `INFO: Linked dose qualifier with code ${uc5} to dose strength with code ${uc6}`
-              );
-            }
-          }
-          // if strength doesn't exist but unit of presentation does
-          else if (uc7) {
-            // link dose qualification to unit of presentation
-            if (!this.graph[uc5].children.map((child) => child.code).includes(uc7)) {
-              this.graph[uc5].children.push({ code: uc7 });
-              console.log(
-                `INFO: Linked dose qualifier with code ${uc5} to unit of presentation with code ${uc7}`
+                `INFO: Linked ${parent.type} with code ${parent.code} to ${child.type} with code ${child.code}`
               );
             }
+            parentIndex = childIndex;
           }
-        }
-
-        // If dose form code exists...
-        if (uc4) {
```

Comment:
I can't see this in the new version

Reply:
_Shouuuuld_ be covered by the above - (happy for feedback on the crude logic though!)